### PR TITLE
Alpine, Arch Linux, Fedora, RHEL, NixOS Root on ZFS: add CI/CD tests

### PR DESCRIPTION
**Pull request by ne9z**
Also move parts of Arch Linux guide to Arch Linux wiki and added Boot Environment page with CI/CD test coverage.

All tests finish successfully, as can be seen [here](https://github.com/ne9z/openzfs-docs/actions/runs/4647024578/jobs/8223800977).

Phew! That was a lot of work.  Hopefully this effort can bring the reliability of the guides to a new level.

### CODE REVIEW

**Comment by ne9z** [comment]
For anyone interesed, here is the complete log of the test run: [logs_108.zip](https://github.com/openzfs/openzfs-docs/files/11184236/logs_108.zip).  It is quite detailed and for most part identical to an installation on a physical machine, which can help us to track down issues.

**Comment by ne9z** [comment]
Enabled shellcheck on all scripts, with the following arguments:
``` sh
shellcheck \
    --check-sourced \
    --enable=all \
    --shell=dash \
    --severity=style \
    --format=tty \
    "${distro}".sh
```

File: docs/Getting Started/Arch Linux/index.rst [hunk 1]
```diff
@@ -31,34 +31,18 @@ ZFS support is provided by third-party `archzfs repo <https://github.com/archzfs
...
-#. Install `zfs-linux* <1-zfs-linux.html>`__
-   or `zfs-dkms <2-zfs-dkms.html>`__ depending on your needs.
-   See the respective pages for details.
-
-Live image
-----------
-Kernel package shipped with latest live image might
-not be compatible with ZFS, user should check kernel version
-following instructions `here <3-live.html>`__.
+See Archlinux Wiki.
```
Comment (gmelikov) [inline docs/Getting Started/Arch Linux/index.rst:34]: Add link to wiki's page?

File: .github/workflows/test_zfs_root_guide.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+name: "Test installation guides"
+on:
+  push:
```
Comment (gmelikov) [inline .github/workflows/test_zfs_root_guide.yml:3]: Let's add path filters to start this workflow only when these root-on-zfs guides were changed https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Just to minimize risks, so we'll have working CI even if these tests have any problems.

Comment (ne9z) [reply .github/workflows/test_zfs_root_guide.yml:3]: Thanks for the link!  I also wanted to limit the tests to only my guides, however I didn't find the related github action doc page.

File: docs/Getting Started/Arch Linux/index.rst (hunk 1 above)
Comment (ne9z) [reply docs/Getting Started/Arch Linux/index.rst]: Added.

**Comment by ne9z** [comment]
I think I will not make any further changes now. But let's wait for a few days.

If there is no more activity in a week, then I will squash the commits.

**Comment by gmelikov** [comment]
@ne9z I've marked this PR as a draft, feel free to remove it when it's ready.

**Comment by ne9z** [comment]
Thank you!

**Comment by ne9z** [comment]
@gmelikov It's ready for review now.  I've also set up daily test in my fork, so that I will receive notifications if something isn't working.

**Comment by gmelikov** [comment]
Sorry for delay, I'll try to review it this week.

**Comment by ne9z** [comment]
Seems NetBSD man page is inaccessible:
```
2023-04-18T15:48:14.5218452Z DEBUG:root:Work on NetBSD...
2023-04-18T15:48:14.5221157Z DEBUG:root:Get http://cvsweb.netbsd.org/bsdweb.cgi/~checkout~/src/external/cddl/osnet/dist/cmd/zpool/zpool-features.7?content-type=text/plain&only_with_tag=MAIN...
2023-04-18T15:48:18.1788133Z Traceback (most recent call last):
2023-04-18T15:48:18.1788504Z   File "/usr/lib/python3.10/urllib/request.py", line 1348, in do_open
2023-04-18T15:48:18.1843367Z     h.request(req.get_method(), req.selector, req.data, headers,
2023-04-18T15:48:18.1844078Z   File "/usr/lib/python3.10/http/client.py", line 1282, in request
2023-04-18T15:48:18.1873755Z     self._send_request(method, url, body, headers, encode_chunked)
2023-04-18T15:48:18.1874197Z   File "/usr/lib/python3.10/http/client.py", line 1328, in _send_request
2023-04-18T15:48:18.1874579Z     self.endheaders(body, encode_chunked=encode_chunked)
2023-04-18T15:48:18.1874959Z   File "/usr/lib/python3.10/http/client.py", line 1277, in endheaders
2023-04-18T15:48:18.1875680Z     self._send_output(message_body, encode_chunked=encode_chunked)
2023-04-18T15:48:18.1876074Z   File "/usr/lib/python3.10/http/client.py", line 1037, in _send_output
2023-04-18T15:48:18.1876376Z     self.send(msg)
2023-04-18T15:48:18.1876660Z   File "/usr/lib/python3.10/http/client.py", line 975, in send
2023-04-18T15:48:18.1876949Z     self.connect()
2023-04-18T15:48:18.1877251Z   File "/usr/lib/python3.10/http/client.py", line 941, in connect
2023-04-18T15:48:18.1877578Z     self.sock = self._create_connection(
2023-04-18T15:48:18.1877922Z   File "/usr/lib/python3.10/socket.py", line 845, in create_connection
2023-04-18T15:48:18.1878215Z     raise err
2023-04-18T15:48:18.1878505Z   File "/usr/lib/python3.10/socket.py", line 833, in create_connection
2023-04-18T15:48:18.1878809Z     sock.connect(sa)
2023-04-18T15:48:18.1879120Z ConnectionRefusedError: [Errno 111] Connection refused
2023-04-18T15:48:18.1879331Z 
```

**Comment by gmelikov** [comment]
@ne9z 

Looks like NetBSD just blocked too much requests, retry was ok.

Please rebase and I'll merge it as is, scripts and CI LGTM, I won't review instructions thoroughly. Let's try it.

**Comment by ne9z** [comment]
> Please rebase and I'll merge it as is, scripts and CI LGTM

Rebased and squashed.  Thanks!

**Pull request merged**